### PR TITLE
Update stress test template to handle Scenario name and group deployments

### PR DESCRIPTION
**Pull request by benbp**
This contains incremental updates to the stress-test-addons helm library:

- Adds an optionally injected scenario name to the global scope, to address unique resource naming conflicts when multiple jobs are deployed as part of a single stress test package
- Changes the `azure-deployer` deployment model to use `az deployment group` instead of `az deployment sub`

Starting as a draft PR because I'm about to go on vacation and I don't want to lose track of changes that https://github.com/Azure/azure-sdk-for-js/pull/16689 is dependent on. However, there is currently a bug where the templates break if `Scenario` is not specified (I think the `printf` function is causing it to evaluate despite the use of `default`).

### CODE REVIEW

File: eng/common/scripts/stress-testing/deploy-stress-tests.ps1 [hunk 1]
```diff
@@ -131,13 +136,13 @@ function DeployStressPackage(
             $imageTag = "${registry}.azurecr.io/$($repository.ToLower())/$($imageName):$deployId"
             Write-Host "Building and pushing stress test docker image '$imageTag'"
             Run docker build -t $imageTag -f $dockerFile.FullName $dockerFile.DirectoryName
-            if ($LASTEXITCODE) { return $LASTEXITCODE }
+            if ($LASTEXITCODE) { return }
```
Comment (weshaggard) [inline eng/common/scripts/stress-testing/deploy-stress-tests.ps1:139]: Do you want to make errors a no-op now?

Comment (benbp) [reply eng/common/scripts/stress-testing/deploy-stress-tests.ps1:139]: I realized this returned exit code isn't used anywhere. The `Run` function will handle adding the failures to a failures array which is then checked and reported after all stress tests have been deployed. I still want the `return` statements for control flow to stop execution on a per-test basis.

File: tools/stress-cluster/cluster/kubernetes/stress-test-addons/templates/_init_deploy.tpl [hunk 2]
```diff
@@ -7,13 +7,19 @@
       source /mnt/secrets/static/* &&
       az login --service-principal -u $AZURE_CLIENT_ID -p $AZURE_CLIENT_SECRET --tenant $AZURE_TENANT_ID &&
       az account set -s $AZURE_SUBSCRIPTION_ID &&
-      az deployment sub create \
-          -n {{ .Release.Name }} \
-          -l westus2 \
+      groupName='{{ lower .Scenario }}-{{ .Release.Name }}-{{ .Release.Revision }}'
+      az group create -l westus2 -g $groupName &&
+      group=$(az group show -g $groupName -o tsv --query "id") &&
+      az tag create --resource-id $group --tags DeleteAfter="$(date -d '+192:00:00' -Iseconds -u)" &&
```
Comment (benbp) [inline tools/stress-cluster/cluster/kubernetes/stress-test-addons/templates/_init_deploy.tpl:13]: As this is growing in complexity, I plan to converge with our common deployment scripts for live tests: https://github.com/Azure/azure-sdk/issues/3211

**Review by weshaggard — APPROVED** [review]

**Pull request merged**